pkg/tombstone/tombstone.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package tombstone
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"time"
+
+	"github.com/cespare/xxhash"
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+
+	"github.com/thanos-io/thanos/pkg/objstore"
+)
+
+const (
+	// TombstoneDir is the name of directory to upload tombstones.
+	TombstoneDir = "tombstones"
+)
+
+// Tombstone represents a tombstone.
+type Tombstone struct {
+	Matcher      string `json:"matcher"`
```

Comment:
I think we need slice here

Reply:
We will be slicing the `matchers` during store API masking and while tombstone creation, we store it as a string. :slightly_smiling_face: